src/Stryker.Core/Stryker.Core/Mutators/MathMutator.cs

```diff
@@ -0,0 +1,146 @@
+
+namespace Stryker.Core.Mutators
+{
+    /// <summary> Mutator Implementation for Math Mutations </summary>
+    public class MathMutator : MutatorBase<ExpressionSyntax>, IMutator
+    {
+        public override MutationLevel MutationLevel => MutationLevel.Advanced;
+
+        /// <summary> Dictionary which maps original Math expressions to the target mutation </summary>
+        private static Dictionary<MathExpression, IEnumerable<MathExpression>> KindsToMutate { get; }
+
+        static MathMutator() => KindsToMutate = new()
+        {
+            [MathExpression.Acos] = new[] { MathExpression.Acosh, MathExpression.Asin, MathExpression.Atan },
+            [MathExpression.Acosh] = new[] { MathExpression.Acos, MathExpression.Asinh, MathExpression.Atanh },
+            [MathExpression.Asin] = new[] { MathExpression.Asinh, MathExpression.Acos, MathExpression.Atan },
+            [MathExpression.Asinh] = new[] { MathExpression.Asin, MathExpression.Acosh, MathExpression.Atanh },
+            [MathExpression.Atan] = new[] { MathExpression.Atanh, MathExpression.Acos, MathExpression.Asin },
+            [MathExpression.Atanh] = new[] { MathExpression.Atan, MathExpression.Acosh, MathExpression.Asinh },
+            [MathExpression.BitDecrement] = new[] { MathExpression.BitIncrement },
+            [MathExpression.BitIncrement] = new[] { MathExpression.BitDecrement },
+            [MathExpression.Ceiling] = new[] { MathExpression.Floor },
+            [MathExpression.Cos] = new[] { MathExpression.Cosh, MathExpression.Sin, MathExpression.Tan },
+            [MathExpression.Cosh] = new[] { MathExpression.Cos, MathExpression.Sinh, MathExpression.Tanh },
+            [MathExpression.Exp] = new[] { MathExpression.Log },
+            [MathExpression.Floor] = new[] { MathExpression.Ceiling },
+            [MathExpression.Log] = new[] { MathExpression.Exp, MathExpression.Pow },
+            [MathExpression.MaxMagnitude] = new[] { MathExpression.MinMagnitude },
+            [MathExpression.MinMagnitude] = new[] { MathExpression.MaxMagnitude },
+            [MathExpression.Pow] = new[] { MathExpression.Log },
+            [MathExpression.ReciprocalEstimate] = new[] { MathExpression.ReciprocalSqrtEstimate },
+            [MathExpression.ReciprocalSqrtEstimate] = new[] { MathExpression.ReciprocalEstimate, MathExpression.Sqrt },
+            [MathExpression.Sin] = new[] { MathExpression.Sinh, MathExpression.Cos, MathExpression.Tan },
+            [MathExpression.Sinh] = new[] { MathExpression.Sin, MathExpression.Cosh, MathExpression.Tanh },
+            [MathExpression.Tan] = new[] { MathExpression.Tanh, MathExpression.Cos, MathExpression.Sin },
+            [MathExpression.Tanh] = new[] { MathExpression.Tan, MathExpression.Cosh, MathExpression.Sinh }
+        };
+
+        /// <summary> Apply mutations to an <see cref="InvocationExpressionSyntax"/> </summary>
+        public override IEnumerable<Mutation> ApplyMutations(ExpressionSyntax node)
```

Comment:
Can't we target a more specific Syntax type? This mutator will now be called for almost every type of node...

Reply:
Thank you, I will limit method to `InvocationExpressionSyntax` during refactoring.